src/main/java/graphql/GraphQLError.java

```diff
@@ -22,19 +26,79 @@
      */
     List<SourceLocation> getLocations();
 
+    /**
+     * @return an enum classifying this error
+     */
     ErrorType getErrorType();
 
+    /**
+     * The graphql spec says that the (optional) path field of any error should be a list
+     * of path entries - http://facebook.github.io/graphql/#sec-Errors
+     *
+     * @return the path in list format
+     */
+    default List<Object> getPath() {
+        return null;
+    }
+
+    /**
+     * The graphql specification says that result of a call should be a map that follows certain rules on what items
+     * should be present.  Certain JSON serializers may or may interpret the error to spec, so this method
+     * is provided to produce a map that strictly follows the specification.
+     *
+     * See : <a href="http://facebook.github.io/graphql/#sec-Errors">http://facebook.github.io/graphql/#sec-Errors</a>
+     *
+     * @return a map of the error that strictly follows the specification
+     */
+    default Map<String, Object> toSpecification() {
+        return Helper.toSpecification(this);
+    }
+
+    /**
+     * @return a map of error extensions or null if there are none
+     */
+    default Map<String, Object> getExtensions() {
+        return null;
+    }
+
 
     /**
      * This little helper allows GraphQlErrors to implement
-     * common things (hashcode/ equals) more easily
+     * common things (hashcode/ equals ) and to specification more easily
      */
     @SuppressWarnings("SimplifiableIfStatement")
     class Helper {
```

Comment:
I don't like this helper inside an interface. Can we extract it to its own class/file?

Reply:
sure